File: LogicalThinkingPatterns/Pyramid/Pyramid.ts
class Pyramid {
  printPattern(n: number): void {
    for (let i = 0; i < n; i++) {
      let pattern: string = "";
      for (let j = n - i - 1; j > 0; j--) {
        pattern += " ";
      }
      for (let k = 0; k <= i; k++) {
        pattern += "* ";
      }
      console.log(pattern);
    }
  }
}

const starPyramid = new Pyramid();
starPyramid.printPattern(5);

File: LogicalThinkingPatterns/Pyramid/Pyramid.py
class Pyramid:
    def printPattern(self,n):
        for i in range(n):
            pattern = ""
            for j in range(n - i - 1, 0, -1):
                pattern += " "
            for k in range(0, i+1):
                pattern +="* "
            print(pattern)
    
    def run(self):
        n = int(input("Enter the number of rows: "))
        self.printPattern(n)

starPyramid = Pyramid()
starPyramid.run()
